drivers/pwm/pwm_cc23x0_timer.c

```diff
@@ -0,0 +1,167 @@
+
+	/* Activate LGPT */
+	HWREG(config->base + LGPT_O_STARTCFG) = 0x1;
+
+	return 0;
+}
+
+static int pwm_cc23x0_get_cycles_per_sec(const struct device *dev, uint32_t channel,
+					 uint64_t *cycles)
+{
+	struct pwm_cc23x0_data *data = dev->data;
+
+	*cycles = data->base_clk / (data->prescale + 1);
+
+	return 0;
+}
+
+static const struct pwm_driver_api pwm_cc23x0_driver_api = {
+	.set_cycles = pwm_cc23x0_set_cycles,
+	.get_cycles_per_sec = pwm_cc23x0_get_cycles_per_sec,
+};
+
+static int pwm_cc23x0_activate_clock(const struct device *dev)
+{
+	const struct pwm_cc23x0_config *config = dev->config;
+	struct pwm_cc23x0_data *data = dev->data;
+	uint32_t lgpt_clk_id = 0;
+
+	switch (config->base) {
+	case LGPT0_BASE:
+		lgpt_clk_id = CLKCTL_LGPT0;
+		break;
+	case LGPT1_BASE:
+		lgpt_clk_id = CLKCTL_LGPT1;
+		break;
+	case LGPT2_BASE:
+		lgpt_clk_id = CLKCTL_LGPT2;
+		break;
+	case LGPT3_BASE:
+		lgpt_clk_id = CLKCTL_LGPT3;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	CLKCTLEnable(CLKCTL_BASE, lgpt_clk_id);
+	HWREG(config->base + LGPT_O_PRECFG) = LGPT_CLK_PRESCALE(data->prescale);
+	HWREG(EVTSVT_BASE + EVTSVT_O_LGPTSYNCSEL) = EVTSVT_LGPTSYNCSEL_PUBID_SYSTIM0;
+
+	return 0;
+}
+
+#define DT_TIMER(idx)           DT_INST_PARENT(idx)
+#define DT_TIMER_BASE_ADDR(idx) (DT_REG_ADDR(DT_TIMER(idx)))
+
+#define PWM_CC23X0_INIT_FUNC(idx)                                                                  \
+	static int pwm_cc23x0_init##idx(const struct device *dev)                                  \
+	{                                                                                          \
+		const struct pwm_cc23x0_config *config = dev->config;                              \
+                                                                                                   \
+		int ret;                                                                           \
+                                                                                                   \
+		LOG_DBG("PWM cc23x0 base=[%x]", config->base);                                     \
+                                                                                                   \
+		ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);                    \
+		if (ret < 0) {                                                                     \
+			LOG_ERR("[ERR] failed to setup PWM pinctrl");                              \
+			return ret;                                                                \
+		}                                                                                  \
+                                                                                                   \
+		pwm_cc23x0_activate_clock(dev);                                                    \
+                                                                                                   \
+		return 0;                                                                          \
+	}
```

Comment:
Am I missing something or is the body of the function exactly the same for all instances? Couldn't just `pwm_cc23x0_init` be used instead?

Reply:
You are correct , this is not needed. Most likely it is just leftover from development. Fixed now.